app/models/printer.rb

```diff
@@ -19,4 +19,19 @@ class Printer < ApplicationRecord
       filters
     end
   end
+
+  after_save :update_printer_in_cups
+
+  def update_printer_in_cups
+    existing_printers = `lpstat -a`.split("\n")
+    existing_printers.map! { |printer| printer.split(' ').first }
+    check_if_printer_exists(existing_printers)
+  end
+
+  def check_if_printer_exists(existing_printers)
+    return if existing_printers.include? name
+    return unless Rails.configuration.auto_create_printer_in_cupsd
+
+    `sudo lpadmin -p #{name} -v socket://#{name}.internal.sanger.ac.uk -E`
+  end
```

Comment:
Unsure about this but probably my lack of understanding!
The `update_printer_in_cups` queries cups and gets all the printers
The `check_if_printer_exists` checks that all those printers we have got from cups exist in cups? So I think they always will?

Does it need to access the PMB database and get all the printers from there, and if they dont exist in cups, then go create them?

Reply:
Yes that is the idea I believe, to make sure any printer existing in PMB database also exists in CUPS

Reply:
`return unless Rails.configuration.auto_create_printer_in_cupsd` can be earlier